### PR TITLE
Make indexer tests insensitive to AZUL_INDEXER_CONCURRENCY (#2107)

### PR DESCRIPTION
**Pull request by jessebrennan**
Author

- [x] PR title references issue
- [x] Title of main commit references issue
- [x] PR is linked to Zenhub issue

Author (reindex)

- [x] Added `r` tag to commit title                         <sub>or this PR does not require reindexing</sub>
- [x] Added `reindex` label to PR                           <sub>or this PR does not require reindexing</sub>

Author (freebies & chains)

- [x] Freebies are blocked on this PR                       <sub>or there are no freebies in this PR</sub>
- [x] Freebies are referenced in commit titles              <sub>or there are no freebies in this PR</sub>
- [x] This PR is blocked by previous PR in chain            <sub>or this PR is not chained to another PR</sub>
- [x] Added `chain` label to the blocking PR                <sub>or this PR is not chained to another PR</sub>

Author (upgrading)

- [x] Documented upgrading of deployments in UPGRADING.rst  <sub>or this PR does not require upgrading</sub>
- [x] Added `u` tag to commit title                         <sub>or this PR does not require upgrading</sub>
- [x] Added `upgrade` label to PR                           <sub>or this PR does not require upgrading</sub>

Author (requirements, before every review)

- [x] Ran `make requirements_update`                        <sub>or this PR leaves requirements*.txt, common.mk and Makefile untouched</sub>
- [x] Added `R` tag to commit title                         <sub>or this PR leaves requirements*.txt untouched</sub>
- [x] Added `reqs` label to PR                              <sub>or this PR leaves requirements*.txt untouched</sub>

Author (before every review)

- [x] `make integration_test` passes in personal deployment <sub>or this PR does not touch functionality that could break the IT</sub>
- [x] Rebased branch on `develop`, squashed old fixups

Primary reviewer (before pushing merge commit)

- [x] Checked `reindex` label and `r` commit title tag
- [x] Rebased and squashed branch
- [x] Sanity-checked history
- [x] Build passes in `sandbox`                             <sub>or added `no sandbox` label</sub>
- [x] Reindexed `sandbox`                                   <sub>or this PR does not require reindexing `sandbox`</sub>
- [x] Added PR reference to merge commit
- [x] Moved linked issue to Merged
- [ ] Pushed merge commit to Github

Primary reviewer (after pushing merge commit)

- [ ] Moved freebies to Merged column                       <sub>or there are no freebies in this PR</sub> 
- [ ] Shortened chain                                       <sub>or this PR is not the base of another PR</sub>
- [ ] Verified that `N reviews` labelling is accurate
- [ ] Commented on demo expectations                        <sub>or labeled as `no demo`</sub>
- [ ] Pushed merge commit to Gitlab                         <sub>or this changes can be pushed later, together with another PR</sub>
- [ ] Deleted PR branch from Github and Gitlab

Primary reviewer (reindex) 

- [ ] Started reindex in `dev`                              <sub>or this PR does not require reindexing `dev`</sub>
- [ ] Deleted BQ slot committment in `dev`                  <sub>or this PR does not require reindexing `dev`</sub>
- [ ] Checked for failures in `dev`                         <sub>or this PR does not require reindexing `dev`</sub>
- [ ] Started reindex in `prod`                             <sub>or this PR does not require reindexing `prod`</sub>
- [ ] Deleted BQ slot committment in `prod`                 <sub>or this PR does not require reindexing `prod`</sub>
- [ ] Checked for failures in `prod`                        <sub>or this PR does not require reindexing `prod`</sub>

Primary reviewer

- [ ] Unassign PR from reviewer and assign to PM

### CODE REVIEW

**Review by hannes-ucsc — CHANGES_REQUESTED** [review]
I would like to see proof that the fix works. You might also discuss this with @amarjandu who recently stumbled upon, and fixed, another non-determinism in the index response, with `_sort`. He might also be good candidate for doing a peer review.

No bump.

**Comment by hannes-ucsc** [comment]
The proof does not need to be permanent. A sequence of separately pushed commit, with or without `drop!` is sufficient.

The reason I mention Amar is because his fix looks simpler.

**Comment by hannes-ucsc** [comment]
A little bit simpler.

**Review by amarjandu — CHANGES_REQUESTED** [review]
I think it might be worth noting about the relation between the `_score` and the `sort`. 

When the query.sort value is set to `_doc` ES attempts to sort values by their position in the  [index.](https://www.elastic.co/guide/en/elasticsearch/reference/6.3/search-request-sort.html#search-request-sort) . The hits.sort appears to indicate the position of the sort field within the index. ( Which _might_  have been introduced when `search_type=scan` was [deprecated](https://www.elastic.co/guide/en/elasticsearch/reference/2.1/breaking_21_search_changes.html#_literal_search_type_scan_literal_deprecated).)

The sort position is no longer relevant because we are forcing the hits to be returned in order of the documents insertion to the index. I believe that the _score in this case is the relevance of the document to the query that was requested, I don't think that we care about the score being returned within the searched we perform here.

Please let me know if there was in error in the review comments I made, I find ES documentation to be difficult to get through, and I feel like at times the information has to be pieced together by referencing different pages of documentation.  🚀

**Comment by amarjandu** [comment]
@jessebrennan How embarrassing I realized that this was not in my review stack to look at. The comments are unwarranted, ignore them if you want.

File: test/indexer/test_hca_indexer.py [hunk 1]
```diff
@@ -121,7 +121,14 @@ class TestHCAIndexer(IndexerTestCase):
     def _get_all_hits(self):
         hits = list(scan(client=self.es_client,
                          index=','.join(self.index_service.index_names(self.catalog)),
-                         doc_type="doc"))
+                         doc_type="doc",
+                         # Without preserve_order, hits are sorted based on
+                         # "_doc", which is fastest. Unfortunately, hit are
+                         # also contaminated with a "sort" field that varies
+                         # unpredictably based on the number of shards. This
```
[the remark this answers is not in the document]
Comment (jessebrennan) [reply test/indexer/test_hca_indexer.py:128]: My statement doesn't preclude that. In the experience of our testing though, only changing the number of shards has caused this test to fail. I'm going to leave this as is.

[the remark this answers is not in the document]
Comment (hannes-ucsc) [reply test/indexer/test_hca_indexer.py:128]: > My statement doesn't preclude that

If we know of another source of non-determinism, we should explicitly mention it.

> In the experience of our testing though, only changing the number of shards has caused this test to fail.

Your testing indicated a dependency on the # of shards. IIRC, @amarjandu's testing suggested another non-determinism, without modifying the # of shards.

**Review by hannes-ucsc — APPROVED** [review]
No bump.

**Comment by hannes-ucsc** [comment]
For demo, disable the fix run a test with different indexer concurrency. Should fail. Then reenable fix and run test again. Should pass.

**Comment by hannes-ucsc** [comment]
Skip pushing to Gitlab. Test changes only.

**Pull request merged**